src/emucore/Console.cxx

```diff
@@ -463,8 +463,8 @@ void Console::initializeAudio()
 */
 void Console::fry() const
 {
-  for (int ZPmem=0; ZPmem<0x100; ZPmem += rand() % 4)
-    mySystem->poke(ZPmem, mySystem->peek(ZPmem) & (uInt8)rand() % 256);
+  for (int ZPmem=0; ZPmem<0x100; ZPmem += randNumGen.next() % 4)
+    mySystem->poke(ZPmem, mySystem->peek(ZPmem) & (uInt8)randNumGen.next() % 256);
```

Comment:
I don't think so. When I run this code:

#include <iostream>
#include <random>

int main(){
std::mt19937 mt_rand(0);
std::cout << mt_rand() << std::endl;
std::cout << (uint8_t) mt_rand() << std::endl;
std::cout << (uint8_t) mt_rand() % 256 << std::endl;
}

I get the following output:

2357136044
/
117

Therefore, the modulo 256 is necessary. Also, I kept what was before, this
modulo and typecasting was already in the code.

On 24 May 2015 at 11:04, Marc G. Bellemare <notifications@github.com> wrote:

> In src/emucore/Console.cxx
> <https://github.com/mgbellemare/Arcade-Learning-Environment/pull/55#discussion_r30953889>
> :
>
> > @@ -463,8 +463,8 @@ void Console::initializeAudio()
> >  */
> >  void Console::fry() const
> >  {
> > -  for (int ZPmem=0; ZPmem<0x100; ZPmem += rand() % 4)
> > -    mySystem->poke(ZPmem, mySystem->peek(ZPmem) & (uInt8)rand() % 256);
> > +  for (int ZPmem=0; ZPmem<0x100; ZPmem += randNumGen.next() % 4)
> > +    mySystem->poke(ZPmem, mySystem->peek(ZPmem) & (uInt8)randNumGen.next() % 256);
>
> Isn't casting next() to uInt8 sufficient to guarantee a modulo 256? I'm
> not sure what the order of operations is here...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mgbellemare/Arcade-Learning-Environment/pull/55/files#r30953889>
> .
>



-- 
Marlos Cholodovskis Machado
http://www.dcc.ufmg.br/~marlos <http://www.cs.ualberta.ca/~machado>